iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/OafBodyWithOrderedUpdates.java

```diff
@@ -0,0 +1,64 @@
+package eu.dnetlib.iis.wf.importer.infospace;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.SortedMap;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+
+import eu.dnetlib.data.proto.OafProtos.Oaf;
+
+/**
+ * {@link Oaf} body along with list of ordered updates which can be applied over this body.
+ * 
+ * @author mhorst
+ *
+ */
+public class OafBodyWithOrderedUpdates {
+
+    private static final String BODY_QUALIFIER_NAME = "body";
+    
+    private String body;
+
+    private Collection<String> orderedUpdates;
+
+    // ------------------------ GETTERS --------------------------
+    
+    /**
+     * Returns body part.
+     */
+    public String getBody() {
+        return body;
+    }
+    
+    /**
+     * Returns all update parts ordered by qualifier.
+     */
+    public Collection<String> getOrderedUpdates() {
+        return orderedUpdates;
+    }
+    
+    // ------------------------ LOGIC --------------------------
+    
+    /**
+     * @param infoSpaceRecords list of {@link QualifiedOafJsonRecord} entities 
+     */
+    public OafBodyWithOrderedUpdates(List<QualifiedOafJsonRecord> infoSpaceRecords) {
+        Preconditions.checkNotNull(infoSpaceRecords);
+        // to be instantiated later because in most cases updates are not available
+        SortedMap<String, String> updatesOrderedByQualifier = null;
```

Comment:
I am not sure what is the real cost but I know updates are defined only for small fraction of millions of records. 

Hadoop reuses `Text` objects to avoid excessive instantiation what may suggest avoiding maps creation is quite reasonable.